client/extensions/woocommerce/app/settings/shipping/shipping-zone-list.js

```diff
@@ -10,74 +12,100 @@ import { localize } from 'i18n-calypso';
+import { getLink } from 'woocommerce/lib/nav-utils';
+import { getShippingZones } from 'woocommerce/state/ui/shipping/zones/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
 
 class ShippingZoneList extends Component {
-	constructor( props ) {
-		super( props );
+	componentWillMount() {
+		if ( this.props.siteId ) {
+			this.props.actions.fetchShippingZones( this.props.siteId );
+		}
+	}
 
-		//TODO: use redux state with real data
-		this.state = {
-			showDialog: false,
-			shippingZones: [ {
-				locationName: 'United States',
-				locationDescription: '50 states',
-				methods: [ {
-					name: 'USPS',
-					description: 'All domestic services',
-				}, {
-					name: 'Flat Rate',
-					description: 'Minimum spend: $100',
-				} ],
-				icon: 'location'
-			}, {
-				locationName: 'Rest of the world',
-				locationDescription: '240 countries',
-				methods: [ {
-					name: 'USPS',
-					description: 'All international services',
-				} ],
-				icon: 'globe'
-			} ],
-		};
+	componentWillReceiveProps( { siteId } ) {
+		if ( siteId === this.props.siteId ) {
+			return;
+		}
+
+		this.props.actions.fetchShippingZones( siteId );
 	}
 
-	renderShippingZone( zone, index ) {
-		return ( <ShippingZone key={ index } { ...zone } /> );
+	renderContent() {
+		if ( ! this.props.loaded ) {
+			return (
+				<div className="shipping__loading-spinner">
+					<Spinner size={ 24 } />
+				</div>
+			);
+		}
+
+		const { translate, siteId } = this.props;
+
+		const renderShippingZone = ( zone, index ) => {
+			return ( <ShippingZoneEntry key={ index } siteId={ siteId } { ...zone } /> );
+		};
+
+		return (
+			<div>
+				<div className="shipping__zones-row shipping__zones-header">
+					<div className="shipping__zones-row-icon"></div>
+					<div className="shipping__zones-row-location">{ translate( 'Location' ) }</div>
+					<div className="shipping__zones-row-methods">{ translate( 'Shipping methods' ) }</div>
+					<div className="shipping__zones-row-actions" />
+				</div>
+				{ this.props.shippingZones.map( renderShippingZone ) }
+			</div>
+		);
 	}
 
 	render() {
-		const { translate } = this.props;
+		const { site, loaded, translate } = this.props;
 
-		const onAddZoneOpen = () => {
-			this.setState( { showDialog: true } );
-		};
+		const addNewHref = loaded
+			? getLink( '/store/settings/shipping/:site/zone/new', site )
+			: '#';
 
-		const onAddZoneClose = () => {
-			this.setState( { showDialog: false } );
+		const onAddNewClick = ( event ) => {
+			if ( ! loaded ) {
+				event.preventDefault();
+			}
 		};
 
 		return (
 			<div>
 				<ExtendedHeader
 					label={ translate( 'Shipping Zones' ) }
 					description={ translate( 'The regions you ship to and the methods you will provide.' ) }>
-					<Button onClick={ onAddZoneOpen }>{ translate( 'Add zone' ) }</Button>
+					<Button
+						href={ addNewHref }
+						onClick={ onAddNewClick }
+						disabled={ ! loaded }>{
+							translate( 'Add zone' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **9** times:
`translate( 'Add DNS Zone' )` *ES Score: 9*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/14275-shipping-methods-ui)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).